### PR TITLE
[Merged by Bors] - feat: add `Continuous.image_connectedComponentIn_subset`

### PR DESCRIPTION
**Pull request by grunweg**
and a version for homeomorphisms. From sphere-eversion; I'm just submitting things upstream.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Homeomorph.lean [hunk 1]
```diff
@@ -318,6 +318,13 @@ theorem isConnected_preimage {s : Set Y} (h : X ≃ₜ Y) :
     IsConnected (h ⁻¹' s) ↔ IsConnected s := by
   rw [← image_symm, isConnected_image]
 
+theorem image_connectedComponentIn {s : Set X} (h : X ≃ₜ Y) {x : X} (hx : x ∈ s) :
+    h '' connectedComponentIn s x = connectedComponentIn (h '' s) (h x) := by
+  refine' (h.continuous.image_connectedComponentIn_subset hx).antisymm _
```
Comment (ADedecker) [inline Mathlib/Topology/Homeomorph.lean:323]:
```suggestion
  refine (h.continuous.image_connectedComponentIn_subset hx).antisymm ?_
```

File: Mathlib/Topology/Connected/Basic.lean [hunk 2]
```diff
@@ -707,6 +707,13 @@ theorem Continuous.image_connectedComponent_subset [TopologicalSpace β] {f : α
     ((mem_image f (connectedComponent a) (f a)).2 ⟨a, mem_connectedComponent, rfl⟩)
 #align continuous.image_connected_component_subset Continuous.image_connectedComponent_subset
 
+theorem Continuous.image_connectedComponentIn_subset [TopologicalSpace β] {f : α → β} {s : Set α}
```
Comment (ADedecker) [inline Mathlib/Topology/Connected/Basic.lean:710]: Could you also add the `mapsTo` version similar to the lemma below?

**Review by ADedecker — APPROVED** [review]
Thanks!

bors d+

**Comment by grunweg** [comment]
Thanks for the quick review.
bors r+

**Pull request closed without merging**